packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -429,11 +430,14 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
   }
 
   async validatePassword(password: string): Promise<PasswordValidationStatus> {
-    // TODO(chazzy): Implement.
-    return Promise.reject(password);
+    if (!this._getPasswordPolicy()) {
+      await this._updatePasswordPolicy();
+    }
+
+    return this._getPasswordPolicy()!.validatePassword(password);
   }
 
-  _getPasswordPolicy(): PasswordPolicy | null {
+  _getPasswordPolicy(): PasswordPolicyInternal | null {
```

Comment:
maybe consider renaming this to _getPasswordPolicyInternal()

Reply:
Done.